dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/JooqBundle.java

```diff
@@ -18,25 +24,66 @@ public void initialize(Bootstrap<?> bootstrap) {
 
     @Override
     public void run(C configuration, Environment environment) throws Exception {
-        final DataSourceFactory dataSourceFactory = getDataSourceFactory(configuration);
-        final JooqFactory jooqFactory = getJooqFactory(configuration);
-        final Configuration cfg = jooqFactory.build(environment, dataSourceFactory);
-        final JooqHealthCheck healthCheck = new JooqHealthCheck(cfg, dataSourceFactory.getValidationQuery());
+        final LinkedHashMap<String,DataSourceFactory> dataSourceFactoryMap = getDataSourceFactories(configuration);
 
-        environment.healthChecks().register("jooq", healthCheck);
-        environment.jersey().register(new JooqBinder(cfg));
-        environment.jersey().register(new LoggingDataAccessExceptionMapper());
+        for (final Map.Entry<String,DataSourceFactory> dataSourceFactoryEntry : dataSourceFactoryMap.entrySet()) {
+            final String name = dataSourceFactoryEntry.getKey();
+            final DataSourceFactory dataSourceFactory = dataSourceFactoryEntry.getValue();
+
+            final JooqFactory jooqFactory = getJooqFactory(configuration);
+            final Configuration cfg = jooqFactory.build(environment, dataSourceFactory, name);
+            final JooqHealthCheck healthCheck = new JooqHealthCheck(cfg, dataSourceFactory.getValidationQuery());
 
-        this.configuration = cfg;
+            environment.healthChecks().register(name, healthCheck);
+
+            jooqFactoryConfigurationMap.put(name, cfg);
+        }
+
+        environment.jersey().register(new JooqBinder(jooqFactoryConfigurationMap));
+        environment.jersey().register(new LoggingDataAccessExceptionMapper());
     }
 
+    /**
+     * Override this method to use a non-default jOOQ configuration.
+     */
     @Override
     public JooqFactory getJooqFactory(C configuration) {
-        // Override this method to use a non-default jOOQ configuration.
         return new JooqFactory();
     }
 
+    /**
+     * Override this method to use a single database.
+     */
+    @Override
+    public DataSourceFactory getDataSourceFactory(C configuration) {
```

Comment:
This would mean an infinite call stack if neither getDataSourceFactories nor getDataSourceFactory were overridden, since getDataSourceFactories already calls getDataSourceFactory. 

I've deprecated getDataSourceFactory and removed the override from ExampleApp.